controller/eks-cluster-config-handler.go

```diff
@@ -215,10 +215,15 @@ func (h *Handler) OnEksConfigRemoved(_ string, config *eksv1.EKSClusterConfig) (
 		}
 	}
 
-	logrus.Infof("deleting node instance role for config [%s]", config.Name)
-	err = deleteStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName))
-	if err != nil {
-		return config, fmt.Errorf("error deleting worker node stack: %v", err)
+	for _, ng := range config.Spec.NodeGroups {
+		if aws.StringValue(ng.NodeRole) == "" {
+			logrus.Infof("deleting node instance role for config [%s]", config.Name)
+			err = deleteStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName))
+			if err != nil {
+				return config, fmt.Errorf("error deleting worker node stack: %v", err)
+			}
+			break
+		}
```

Comment:
This makes sense @thedadams and is a great catch. I will try to get this done by the end of the week. Would an api call to check if the role exists be a possible solution?

Reply:
These look to be the errors we may receive from requests in `deleteStack` function [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/CommonErrors.html). I dont see any fine grained to not-found but i imagine one of these handles it, would there be any worry if the error fails gracefully on another error that is not a not-found but covered by the same error response so the code would handle it as it would a not-found?